pkg/mic/mic.go

```diff
@@ -627,7 +632,14 @@ func (c *Client) createDesiredAssignedIdentityList(
 					klog.Errorf("failed to create an AzureAssignedIdentity between pod %s/%s and AzureIdentity %s/%s, error: %+v", pod.Namespace, pod.Name, azureID.Namespace, azureID.Name, err)
 					continue
 				}
-				newAssignedIDs[assignedID.Name] = *assignedID
+
+				if a, ok := newAssignedIDs[assignedID.Name]; ok {
```

Comment:
Could we add an e2e for this behavior? 

1. Create `AzureIdentity` with same name in 2 namespaces
2. Create `AzureIdentityBinding`
3. Use the selector in pod and check if the `AzureIdentity` that's picked is always the first one in the sorted list